### DISCUSSION TITLE
Allow blocking of visitor IP ranges / CIDR blocks for github pages properties

### DISCUSSION THREAD

**Discussion by james-see** [community/community#9713] [Other Feature Feedback, Questions, & Ideas]
Hello, I was trying to block an entire ip range based on country from visiting my website that I host on github pages. I added a .htaccess to my root folder hoping Github Pages would somehow interpolate that I wanted those ip addresses blocked, but this did not work since I am assuming the web server would have to be apache and unless specific code was in place to read that config, it would just be ignored.

I thought about writing some javascript to redirect, but they would just have to disable javascript to get through. 

This seems like a feature that should be added to the pages settings tab to paste in a list of CIDR blocks or ip addresses to ban as a nice security feature for known bad-actor ip addresses, etc.

**Comment by jfullstackdev** [comment]
I don't know whether this is answered by the GitHub Staff here

https://github.com/orgs/community/discussions/27813#discussioncomment-3300331

**Comment by WolvenSpirit** [comment]
If your page makes requests to an API you own, you could simply block the content from showing by checking the client IP that makes the request to fetch data from your API endpoint. If it's static page, you can't do much probably unless you use some free service to tell you "your" ip using JS executed client-side. (this https://stackoverflow.com/questions/391979/how-to-get-clients-ip-address-using-javascript)

**Reply by WolvenSpirit** [reply]
Github Pages does not strip in pages of client-side JS scripts so that's your advantage right there to make your IP range filter.
Bad news is anybody can prevent it if they know a bit of code but it's better than nothing.

**Reply by james-see** [reply]
Yeah, I was going to say they could just disable it, but a good stop sign at least.
